R/dataprep_table_formatting.R

```diff
@@ -0,0 +1,80 @@
+#' as_update_existing
+#' 
+#' Converts a sidora table to a format that can be use with pandora table upload
+#' interface
+#' 
+#' @examples
+#' \dontrun{
+#' sidora_table <- get_df("TAB_Site", con)
+#' format_as_update_existing(sidora_table,)
+#' }
+#' 
+#' @param sidora_table a sidora table
+#' 
+#' @rdname format_as_update_existing
+#' @export
+format_as_update_existing <- function(sidora_table) {
+  
+  ## Find valid columns for uploading existing tables
+  valid_cols_table_clean <- valid_cols_table <- hash::values(hash_pandora_col_name_update_type, colnames(sidora_table))
+  names(valid_cols_table_clean) <- sidora_col_name_to_update_col_name(names(valid_cols_table_clean))
+  valid_cols_table_clean_mandatory <- names(valid_cols_table_clean[valid_cols_table_clean %in% "mandatory"])
+  
+  ## For upload 'Full_*_Id' column isn't used, but just '_Id', which here is a 
+  ## numeric value . We will replace the latter with the former here for export
+  ## QUESTION maybe put as data object? But will never be used other than here
+  name_map <- c(individual.Full_Individual_Id = "individual.Individual_Id", 
+                sample.Full_Sample_Id = "sample.Sample_Id", 
+                extract.Full_Extract_Id = "extract.Extract_Id", 
+                library.Full_Library_Id = "library.Library_Id", 
+                capture.Full_Capture_Id = "capture.Capture_Id", 
+                sequencing.Full_Sequencing_Id = "sequencing.Sequencing_Id", 
+                raw_data.Full_Raw_Data_Id = "raw_data.Raw_Data_Id"
+                )
+  
+  string_col <- name_map[names(sidora_table)[names(sidora_table) %in% names(name_map)]]
+  sidora_table[string_col] <- sidora_table[names(string_col)]
+  sidora_table 
+  
+  ## Clean up tables, subsetting and fixing column types when necessary
+  result <- sidora_table[, !is.na(valid_cols_table)] %>% fix_logical_update_existing()
+  colnames(result) <- sidora_col_name_to_update_col_name(names(result))
+  
+  ## Add formatting for mandatory upload columns
+  result %>% dplyr::rename_with(make_column_mandatory, valid_cols_table_clean_mandatory)
+}
+
+#' make_column_mandatory
+#'
+#' @param x column name to wrap with asterisks
+make_column_mandatory <- function(x){
+  gsub("^", "\\*", x) %>% gsub("$", "\\*", .)
+}
+
+#' fix_logical_update_existing
+#' 
+#' @param x column name to wrap with asterisks
+
+fix_logical_update_existing <- function(sidora_table){
+  
```

Comment:
Maybe it would be good to add a comment why these two hacks are necessary. Might be pretty confusing for future readers of this code.